shared-bindings/struct/__init__.c

```diff
@@ -0,0 +1,185 @@
+/*
+ * This file is part of the MicroPython project, http://micropython.org/
+ *
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2013, 2014 Damien P. George
+ * Copyright (c) 2014 Paul Sokolovsky
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#include <assert.h>
+#include <string.h>
+
+#include "py/runtime.h"
+#include "py/builtin.h"
+#include "py/objtuple.h"
+#include "py/binary.h"
+#include "py/parsenum.h"
+#include "shared-bindings/struct/__init__.h"
+
+//| :mod:`struct` --- manipulation of c-style data
+//| ========================================================
+//|
+//| .. module:: struct
+//|   :synopsis: byte data control
+//|   :platform: SAMD21
+//|
+//| This module implements a subset of the corresponding CPython module,
+//| as described below. For more information, refer to the original CPython
+//| documentation: struct.
+//|
+//| Supported size/byte order prefixes: *@*, *<*, *>*, *!*.
+//|
+//| Supported format codes: *b*, *B*, *h*, *H*, *i*, *I*, *l*, *L*, *q*, *Q*,
+//| *s*, *P*, *f*, *d* (the latter 2 depending on the floating-point support).
+
+
+/*
+    This module implements most of character typecodes from CPython, with
+    some extensions:
```

Comment:
Are these still supported? This is the kinda of thing I don't want to support because it won't port to CPython.

Reply:
I don't know of any extensions other than some slight behavior differences that i already have pointed out.  I took the text from the original code, so i will remove the words ,'with some extensions'

Reply:
I looked at the micro python documentation.  They do not document these extensions.  We won't either.  There is some code that is looking for 'S'.  I don't yet see any for 'O'.  I can delete the 'S' code.